python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -1479,94 +1343,84 @@ def _execute_asset_backfill_iteration_inner(
-        # Events are not guaranteed to be written to the event log in monotonic increasing order,
-        # so we wait to ensure all events up until next_latest_storage_id have been written.
-        if cursor_delay_time:
-            time.sleep(cursor_delay_time)
+    updated_materialized_subset = get_asset_backfill_iteration_materialized_subset(
+        backfill_id, asset_backfill_data, asset_graph, instance_queryer
+    )
 
-        updated_materialized_subset = get_asset_backfill_iteration_materialized_subset(
-            backfill_id, asset_backfill_data, asset_graph, instance_queryer
-        )
+    materialized_since_last_tick = (
+        updated_materialized_subset - asset_backfill_data.materialized_subset
+    )
+    logger.info(
+        f"Assets materialized since last tick:\n{_asset_graph_subset_to_str(materialized_since_last_tick, asset_graph)}"
+        if not materialized_since_last_tick.is_empty
+        else "No relevant assets materialized since last tick."
+    )
 
-        materialized_since_last_tick = (
-            updated_materialized_subset - asset_backfill_data.materialized_subset
-        )
-        logger.info(
-            f"Assets materialized since last tick:\n{_asset_graph_subset_to_str(materialized_since_last_tick, asset_graph)}"
-            if not materialized_since_last_tick.is_empty
-            else "No relevant assets materialized since last tick."
-        )
+    failed_asset_graph_subset = _get_failed_asset_graph_subset(
+        asset_graph_view,
+        backfill_id,
+        updated_materialized_subset,
+    )
 
-        failed_asset_graph_subset = _get_failed_asset_graph_subset(
-            asset_graph_view,
-            backfill_id,
-            updated_materialized_subset,
-        )
-        candidate_asset_graph_subset = _get_candidate_asset_graph_subset(
-            asset_backfill_data,
-            asset_graph_view,
-            updated_materialized_subset,
-            failed_asset_graph_subset,
-        )
+    failed_and_downstream_subset = _get_failed_and_downstream_asset_graph_subset(
+        backfill_id,
+        asset_backfill_data,
+        asset_graph_view,
+        updated_materialized_subset,
+        failed_asset_graph_subset,
+    )
 
-        failed_and_downstream_subset = _get_failed_and_downstream_asset_graph_subset(
-            backfill_id,
-            asset_backfill_data,
-            asset_graph_view,
-            updated_materialized_subset,
-            failed_asset_graph_subset,
-        )
+    candidate_asset_graph_subset = (
+        asset_backfill_data.target_subset
+        - asset_backfill_data.requested_subset
+        - failed_and_downstream_subset
+    )
 
     logger.info(
         f"Considering the following candidate subset:\n{_asset_graph_subset_to_str(candidate_asset_graph_subset, asset_graph)}"
         if not candidate_asset_graph_subset.is_empty
         else "Candidate subset is empty."
     )
```

Comment:
we can potentially remove this log line once the candidate subset is more determnistic?

Reply:
sorta nice to have the system do that accounting for you I think, not hard set on that